astropy/io/fits/tests/test_convenience.py

```diff
@@ -157,6 +157,46 @@ def test_tabledump(self):
         fits.tabledump(self.temp('tb.fits'), datafile=self.temp('test_tb.txt'))
         assert os.path.isfile(self.temp('test_tb.txt'))
 
+    def test_append(self):
+        """
+        Test that append works in the case of an existing test file
+        """
+        # Test case 1
+        fits.append('test_out_1.fits', data=np.ones((16,32)), output_verify='fix')
+        hdu1 = fits.open('test_out_1.fits', mode='update', checksum=True)
+        assert hdu1 is not False
+
+        # Test case 2
+        import tempfile
+        fp = tempfile.NamedTemporaryFile(prefix='test_file_', suffix='.fits', dir='.')
+        fits.append(fp, data=np.ones((16,32)), output_verify='fix')
+        assert os.path.exists(fp.name)
+
+        hdu2 = fits.open(fp, mode='update', checksum=True)
+        assert hdu2 is not False
+
+        # Test case 3
+        data = np.ones((3, 4, 5))
+
+        import astropy.wcs
+        wcs_input_dict = {
+        'CTYPE1': 'AWAV    ', 'CUNIT1': 'Angstrom', 'CDELT1': 0.2, 'CRPIX1': 0, 'CRVAL1': 10, 'NAXIS1': 5,
+        'CTYPE2': 'RA---TAN', 'CUNIT2': 'deg', 'CDELT2': 0.5, 'CRPIX2': 2, 'CRVAL2': 0.5, 'NAXIS2': 4,
+        'CTYPE3': 'DEC--TAN', 'CUNIT3': 'deg', 'CDELT3': 0.4, 'CRPIX3': 2, 'CRVAL3': 1, 'NAXIS3': 3
+        }
+        input_wcs = astropy.wcs.WCS(wcs_input_dict)
+        hdr = input_wcs.to_header()
+
+        fits.writeto('test_out_2.fits', data, hdr)
+        fits.append('test_out_2.fits', data=3*np.ones((3, 4, 5)))
```

Comment:
I don't understand the reason to put a wcs here. And this is still not testing the third case ... for this you need to set verify=False.

Reply:
Sorry @saimn, I am still not full grasping the concept of a FITS file object. Would you mind giving me an example or two in order to teach me what it really means? I tried looking it up, but am still unsure, hence my vain attempt. I would appreciate any help in this regard.

Reply:
If you have a more precise question, or something in the documentation that is not clear, please ask. But here I don't know what to explain.

Reply:
Thanks, I think I have confused a FITS HDUList object with a file object earlier, hence the confusion...